src/Components/ArtworkGrid/ArtworkGridContext.tsx

```diff
@@ -15,20 +15,41 @@ interface ArtworkGridContextProps {
   saveOnlyToDefaultList?: boolean
 
   hideSignals?: boolean
+
+  signals?: { [id: string]: string[] }
+
+  updateSignals?: (id: string, newSignals: string[]) => void
```

Comment:
To summarize the approach, the context maintains a `signals` data structure that keeps track of all the artworks and their individual collector signals (which can be updated client side in the `PrimaryLabelLine` component). In the grid/rail item `onClick` callback, we'll refer to this `signals` map to fire tracking calls with the correct signals (which may or may not be overwritten on the client side).